app/controllers/users_controller.rb

```diff
@@ -1,10 +1,35 @@
 class UsersController < ApplicationController
-  skip_before_action :authenticate_user!, only: %i[index show]
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :ensure_chef, only: [:edit, :update]
+
   def index
     @chefs = User.all
   end
 
   def show
     @chef = User.find(params[:id])
   end
+
+  def edit
+    @chef = current_user
+  end
+
+  def update
+    @chef = current_user
+    if @chef.update(chef_params)
+      redirect_to @chef, notice: 'Profile updated successfully!'
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def chef_params
+    params.require(:user).permit(:first_name, :last_name, :bio, :specialties)
```

Comment:
Your attributes here don't match what is in the database, please check the schema and correct them.

Reply:
chnaged to match schema